src/overlays/actors/ovl_Boss_Ganon/z_boss_ganon.c

```diff
@@ -108,127 +98,4921 @@ static ColliderCylinderInit D_808E4C2C = {
+    s16 i;
+    s16 j;
+    ColliderInfo* acHitInfo;
+
+    if (this->collider.base.acFlags & 2) {
+        this->unk_2D4 = 2;
+        this->collider.base.acFlags &= ~2;
+        acHitInfo = this->collider.info.acHitInfo;
+
+        if ((this->actionFunc == BossGanon_HitByLightBall) || (this->actionFunc == BossGanon_ChargeBigMagic)) {
+            if (acHitInfo->toucher.dmgFlags & 0x2000) {
+                BossGanon_SetupVulnerable(this, globalCtx);
+                this->timers[2] = 0;
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_GANON_DAMAGE1);
+                this->unk_1A6 = 15;
+            }
+        } else if ((this->actionFunc == BossGanon_Vulnerable) && (this->unk_1C2 >= 3)) {
+            if (!(acHitInfo->toucher.dmgFlags & 0x80)) {
+                u8 hitWithSword = false;
+                u8 damage;
+                Vec3f sp50;
+                u32 flags;
+
+                for (i = 0; i < 30; i++) {
+                    sp50.x = Rand_CenteredFloat(20.0f);
+                    sp50.y = Rand_CenteredFloat(20.0f);
+                    sp50.z = Rand_CenteredFloat(20.0f);
+                    BossGanonEff_SpawnSparkle(globalCtx, &this->unk_1FC, &sp50, &sZeroVec,
+                                              Rand_ZeroFloat(200.0f) + 500.0f, 0x1E);
+                }
+
+                damage = flags = CollisionCheck_GetSwordDamage(acHitInfo->toucher.dmgFlags);
+
+                if (flags == 0) {
+                    damage = 2;
+                } else {
+                    hitWithSword = true;
+                }
+
+                if (((s8)this->actor.colChkInfo.health >= 3) || hitWithSword) {
+                    this->actor.colChkInfo.health -= damage;
+                }
+
+                for (i = 0; i < ARRAY_COUNT(sCape->strands); i++) {
+                    for (j = 1; j < 12; j++) {
+                        sCape->strands[i].velocities[j].x = Rand_CenteredFloat(15.0f);
+                        sCape->strands[i].velocities[j].z = Rand_CenteredFloat(15.0f);
+                    }
+                }
+
+                if ((s8)this->actor.colChkInfo.health <= 0) {
+                    BossGanon_SetupDeathCutscene(this, globalCtx);
+                    Audio_PlayActorSound2(&this->actor, NA_SE_EN_GANON_DEAD);
+                    Audio_PlayActorSound2(&this->actor, NA_SE_EN_GANON_DD_THUNDER);
+                    func_80078914(&sZeroVec, NA_SE_EN_LAST_DAMAGE);
+                    Audio_QueueSeqCmd(0x100100FF);
+                    this->screenFlashTimer = 4;
+                } else {
+                    Audio_PlayActorSound2(&this->actor, NA_SE_EN_GANON_DAMAGE2);
+                    Audio_PlayActorSound2(&this->actor, NA_SE_EN_GANON_CUTBODY);
+                    BossGanon_SetupDamaged(this, globalCtx);
+                    this->unk_1A6 = 15;
+                    sCape->tearTimer = 1;
+                }
+            }
+        } else if ((acHitInfo->toucher.dmgFlags & 0x1F8A4) != 0) {
```

Comment:
@Dragorn421 suggested the != 0 to be there, so yall will have to discuss that. I dont think i mind either way

Reply:
im removing it since it conforms to our pattern with flag checks in general, but dragorn had a reason for wanting it there which is still good to talk about